streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/base/config/SpringProperties.java

```diff
@@ -87,17 +87,37 @@ private static void dataSourceConfig(Properties userConfig, Properties springCon
         springConfig.put("spring.datasource.driver-class-name", "org.postgresql.Driver");
         break;
       case "h2":
-        springConfig.put("spring.datasource.driver-class-name", "org.h2.Driver");
-        springConfig.put("spring.datasource.username", "sa");
-        springConfig.put("spring.datasource.password", "sa");
+        String h2Dir = userConfig.getProperty("datasource.h2-data-dir", null);
+        if (StringUtils.isBlank(h2Dir)) {
+          h2Dir = System.getProperty("user.home", "~") + "/streampark/metadata";
```

Comment:
I think we should directly using `h2Dir`  value so that can reduce the user's cost of use.

Reply:
> I think we should directly using `h2Dir` value so that can reduce the user's cost of use.

updated.